driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -179,6 +189,64 @@ public static <R> void poolClearedExceptionMustBeRetryable(
         }
     }
 
+    /**
+     * Prose test #3.
+     */
+    @Test
+    public void originalErrorMustBePropagatedIfNoWritesPerformed() throws InterruptedException {
+        originalErrorMustBePropagatedIfNoWritesPerformed(MongoClients::create);
+    }
+
+    public static void originalErrorMustBePropagatedIfNoWritesPerformed(
+            final Function<MongoClientSettings, MongoClient> clientCreator) throws InterruptedException {
+        assumeTrue(serverVersionAtLeast(6, 0) && isDiscoverableReplicaSet());
+        BiFunction<Integer, List<String>, BsonDocument> configureFailPointDocCreator = (errorCode, errorLabels) ->
+                new BsonDocument()
+                        .append("configureFailPoint", new BsonString("failCommand"))
+                        .append("mode", new BsonDocument()
+                                .append("times", new BsonInt32(1)))
+                        .append("data", new BsonDocument()
+                                .append("failCommands", new BsonArray(singletonList(new BsonString("insert"))))
+                                .append("errorCode", new BsonInt32(errorCode))
+                                .append("errorLabels", new BsonArray(errorLabels.stream().map(BsonString::new).collect(Collectors.toList()))));
+        ServerAddress primaryServerAddress = Fixture.getPrimary();
+        CompletableFuture<FailPoint> futureFailPoint = new CompletableFuture<>();
+        CommandListener commandListener = new CommandListener() {
+            private final AtomicBoolean configureFailPoint = new AtomicBoolean(true);
+
+            @Override
+            public void commandFailed(final CommandFailedEvent event) {
```

Comment:
The prose test [says to use `CommandSucceededEvent`](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/tests/README.rst#prose-tests:~:text=Via%20the%20command%20monitoring%20CommandSucceededEvent), but the server responds with `ok: 0` to the first attempt in the test, which means, we need to use `CommandFailedEvent`. [People observer different server behavior when manually executing the test steps in `mongosh`](https://mongodb.slack.com/archives/C72LB5RPV/p1665185092241459?thread_ts=1665183651.355829&cid=C72LB5RPV), but I have not been able to observe the same.

This situation is weird and is likely caused by a difference in the `insert` command that is being sent, though I don't know what that difference is.

Reply:
Let's get to the bottom of this before merging this PR

Reply:
This discrepancy as caused by me creating an incorrect fail point document.